### PR TITLE
displays only orders of a waiting status in delivery summary page

### PR DESCRIPTION
**Pull request by this-username-is-taken**

### CODE REVIEW

File: delivery/views.py [hunk 1]
```diff
@@ -7,6 +7,16 @@
...
 from order.models import DeliveryWindow, Order
 
+import sys
+
+def delivery_window_by_status(window, status):
+  window.order_list = []
+  for order in window.orders.all():
+    if order.status == status:
+      window.order_list.append(order)
+  print >>sys.stderr, window.order_list
```
Comment (gregwym) [inline delivery/views.py]: Debug print shouldn't be committed.

File: delivery/views.py [hunk 2]
```diff
@@ -7,6 +7,16 @@
 
 from order.models import DeliveryWindow, Order
 
+import sys
```
Comment (gregwym) [inline delivery/views.py]: Remove this together with the debug print.

File: delivery/views.py [hunk 3]
```diff
@@ -20,6 +30,8 @@ def divide_delivery_window_by_days(delivery_windows):
 
   datetime_threshold = make_aware(datetime.now() - timedelta(days=30), get_default_timezone())
   delivery_windows = DeliveryWindow.objects.filter(start__gt=datetime_threshold).order_by('-start', 'store__id')
+  for delivery_window in delivery_windows:
+    delivery_window = delivery_window_by_status(delivery_window, Order.STATUS_WAITING)
```
Comment (gregwym) [inline delivery/views.py]: I know its working, other wise you will not commit it. But I don't think its the right way of doing it. 

The problem is, `DeliveryWindow` is a Django Model class, and it doesn't have `order_list` as its property. You are adding a property `order_list` to a `DeliveryWindow` instance. This will results in inconsistent state between the in-memory `delivery_window` object and the one in db. We want to avoid this kinda of behavior. 

To summarize, we don't want to manipulate a `Model` object unless we are going to save the changes to the db.

Comment (gregwym) [inline delivery/views.py]: The proper way to achieve is to add instance method to the `DeliveryWindow` model and add an property for it is even better. 
https://docs.djangoproject.com/en/dev/topics/db/models/#model-methods

Make sure the naming make sense. 

For example, 
https://github.com/Fruitex/fruitex/blob/dev/order/models.py#L137
https://github.com/Fruitex/fruitex/blob/dev/order/models.py#L163

File: ntemplates/delivery/detail.html [hunk 4]
```diff
@@ -9,11 +9,11 @@
 <h1>Delivery Window</h1>
 <h2>{{ delivery_window.store }}: {{ delivery_window }}</h2>
 <ul>
-  {% for order in delivery_window.orders.all %}
+  {% for order in delivery_window.order_list %}
   <li>
     <div>
       <a href="{% url 'order:show' order.invoice.invoice_num %}">
-        Order {{ order }}: {{ order.get_status_display }} ({{ order.items.count }} items)
+        Order {{ order }}: ({{ order.items.count }} items)
```
Comment (gregwym) [inline ntemplates/delivery/detail.html]: Display the order status doesn't hurt. We are going to have status more than 'Pending' and 'Waiting'.

File: ntemplates/delivery/summary.html [hunk 5]
```diff
@@ -14,15 +14,15 @@
...
       </a>
     </div>
     <ul>
-      {% for order in delivery_window.orders.all %}
+      {% for order in delivery_window.order_list %}
       <li>
         <div>
           <a href="{% url 'order:show' order.invoice.invoice_num %}">
-            Invoice #{{ order.invoice.id }} Order #{{ order.id }}: {{ order.get_status_display }} ({{ order.items.count }} items)
+            Invoice #{{ order.invoice.id }} Order #{{ order.id }}: ({{ order.items.count }} items)
```
Comment (gregwym) [inline ntemplates/delivery/summary.html]: Same here.

**Comment by gregwym** [comment]
Don't close a PR unless you are not going to open another one. Just push on your branch, the PR will update automatically.

**Comment by gregwym** [comment]
I guess I should open the first one so we keep track of how was everything going.

**Comment by gregwym** [comment]
Don't reset your commit and force push. Just append commit when you are fixing stuff for a PR. Its better that way so we keep track of how the decisions were made.

**Comment by this-username-is-taken** [comment]
yea, I will do that next time, for some reason it didn't work
